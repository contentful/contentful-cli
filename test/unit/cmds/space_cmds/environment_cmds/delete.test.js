const {
  environmentDelete
} = require('../../../../../lib/cmds/space_cmds/environment_cmds/delete')
const { getContext } = require('../../../../../lib/context')
const {
  createManagementClient
} = require('../../../../../lib/utils/contentful-clients')

jest.mock('../../../../../lib/context')
jest.mock('../../../../../lib/utils/contentful-clients')

const deleteEnvironmentStub = jest.fn()
const environmentData = {
  name: 'environment name',
  sys: {
    id: 'environmentId'
  },
  delete: deleteEnvironmentStub
}
const getEnvironmentStub = jest.fn().mockResolvedValue(environmentData)

const fakeClient = {
  getSpace: async () => ({
    getEnvironment: getEnvironmentStub
  })
}
createManagementClient.mockResolvedValue(fakeClient)

getContext.mockResolvedValue({
  managementToken: 'mockedToken'
})

afterEach(() => {
  createManagementClient.mockClear()
  createManagementClient.mockClear()
  getEnvironmentStub.mockClear()
  deleteEnvironmentStub.mockClear()
})

test('delete environment', async () => {
  const result = await environmentDelete({
    context: {
      activeSpaceId: 'someSpaceID'
    },
    environmentId: 'someEnvironmentID'
  })
  expect(result).toBeTruthy()
  expect(createManagementClient).toHaveBeenCalledTimes(1)
  expect(getEnvironmentStub).toHaveBeenCalledTimes(1)
  expect(getEnvironmentStub.mock.calls[0][0]).toBe('someEnvironmentID')
  expect(deleteEnvironmentStub).toHaveBeenCalledTimes(1)
})
