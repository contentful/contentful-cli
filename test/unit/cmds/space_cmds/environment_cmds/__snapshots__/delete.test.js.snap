// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete environment - requires space id 1`] = `"You need to provide a space id. You can pass it via the --space-id parameter or by running contentful space use"`;
